heron/api/src/java/com/twitter/heron/api/bolt/WindowedBoltExecutor.java

```diff
@@ -258,6 +260,8 @@ private boolean isTupleTs() {
         return new WatermarkTimeTriggerPolicy<>(slidingIntervalDurationMs, manager,
             evictionPolicy, manager);
       } else {
+        // set tick tuple frequency for timer in TimeTriggerPolicy
+        Config.setTickTupleFrequency(topoConf, slidingIntervalDurationMs);
```

Comment:
Instead of updating the config here within a bolt, can we push this task to topology creation time and provide some default value if user doesn't set it? In this way, you don't need to change the `BoltInstance` code and pass around the conf object

Reply:
@nlu90 not all types of windows need the timer.  Only processing time based windows that use the TimerTriggerPolicy need a timer thus tick tuples other window types do not need it.   
 Its not eloquent to determine what kind of window it is on the client side. The existing python windowing implementation also sets the tick tuple frequency in the initialize/prepare method.